### PR TITLE
733 Design Section Revamp

### PR DESCRIPTION
**Pull request by ACStoneCL**
### Related links

[Condense design section - Part 1 #733](https://github.com/casper-network/docs/issues/733)

### Changes

Condensing and revamping the Design section of our documentation.

### Notes

Ran locally without issue - ensured all links and images working correctly.

### CODE REVIEW

**Comment by RitaMAllenCA** [comment]
@ACStoneCL is there any reason this would not merge by EOD MOnday

**Comment by ACStoneCL** [comment]
@RitaMAllenCA I don't see this merging on Monday, if only because I don't want to rush reviews after the meeting with Ed that afternoon and any necessary revisions that emerge.

File: source/docs/casper/design/casper-design.md [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+1. Deploy Received
+2. Deploy Gossiped
+3. Block Proposed
+4. Block Gossiped
+5. Consensus Reached
+6. Deploy Executed
+
+#### Deploy Received
+The client sending the deploy will send it to one or more nodes via their JSON RPC servers. The node will ensure that a given deploy matches configuration settings as set forth in the network's chainspec. Deploy configuration for the Casper Mainnet can be found [here](https://github.com/casper-network/casper-node/blob/dev/resources/production/chainspec.toml#L79). Once accepted, the deploy hash is returned to the client to indicate it has been enqueued for execution. The deploy could expire while waiting to be gossiped and whenever this happens a `DeployExpired` event is emitted by the event stream servers of all nodes which have expired the deploy.
```
Comment (darthsiroftardis) [inline source/docs/casper/design/casper-design.md:110]:
```suggestion
A client sending the deploy will send it to one or more nodes via their JSON RPC servers. The node will ensure that a given deploy matches configuration settings as set forth in the network's chainspec. Deploy configuration for the Casper Mainnet can be found [here](https://github.com/casper-network/casper-node/blob/dev/resources/production/chainspec.toml#L79). Once accepted, the deploy hash is returned to the client to indicate it has been enqueued for execution. The deploy could expire while waiting to be gossiped and whenever this happens a `DeployExpired` event is emitted by the event stream servers of all nodes which have expired the deploy.
```

**Review by darthsiroftardis — APPROVED** [review]

**Review by ipopescu — APPROVED** [review]
Great work, @ACStoneCL! I submitted an editorial pass and left a few comments which can be discussed and addressed now or in a subsequent PR.

**Pull request merged**